core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -161,8 +163,19 @@ private static void toJson(TableMetadata metadata, JsonGenerator generator) thro
     generator.writeNumberField(LAST_UPDATED_MILLIS, metadata.lastUpdatedMillis());
     generator.writeNumberField(LAST_COLUMN_ID, metadata.lastColumnId());
 
-    generator.writeFieldName(SCHEMA);
-    SchemaParser.toJson(metadata.schema(), generator);
+    // for older readers, continue writing the current schema as "schema"
```

Comment:
Might want to note that this is done until v2 because support for `schemas` and `current-schema-id` is required in v2 and later.